generator/src/test/java/com/scottlogic/deg/generator/generation/RegexStringGeneratorTests.java

```diff
@@ -246,6 +248,20 @@ void shouldReturnValuesWhenNonContradictingConstraints() {
         Assert.assertNotEquals(0, nonContradictingGenerator.getValueCount());
     }
 
+    @Test
+    void shouldNotGenerateInvalidUnicodeCodePoints() {
+        StringGenerator generator = new RegexStringGenerator("[😁-😘]{1}", true);
+        Iterable<String> resultsIterable = generator.generateAllValues();
+        for (String s : resultsIterable) {
```

Comment:
Minor point, but is there a way you can reduce the nesting here, the statements go as far as a 5 deep level of nesting

Reply:
I have reduced the nesting, but it may cause performance of the test to be reduced.